django_cloud_tasks/signals.py

```diff
@@ -0,0 +1,88 @@
+# pylint: disable=no-member
+from django.core.exceptions import ValidationError
+from django.db.models.signals import pre_save
+from django.dispatch import receiver
+from django.db.models import Model
+
+from django_cloud_tasks import models, tasks
+
+
+@receiver(pre_save, sender=models.Routine)
+def ensure_valid_task_name(sender, instance, **kwargs):
+    # A exception is raised when the task is not found
+    instance.task  # pylint: disable=pointless-statement
+
+
+def _is_status_changing(instance: Model) -> bool:
+    if not instance.pk:
+        return False
+
+    current_routine = models.Routine.objects.get(pk=instance.pk)
+
+    return current_routine.status != instance.status
+
+
+@receiver(pre_save, sender=models.Routine)
+def enqueue_next_routines(sender, instance, **kwargs):
+    if not _is_status_changing(instance=instance):
+        return
+
+    if instance.status == models.Routine.Statuses.COMPLETED:
+        for vertice in instance.next_routine_vertices.all():
```

Comment:
Use the M2M field to make it more readable:
`for routine in instance.next_routines.all()`